cmd/kubelet/app/server.go

```diff
@@ -126,7 +126,7 @@ func UnsecuredKubeletDeps(s *options.KubeletServer) (*kubelet.KubeletDeps, error
 
 	var dockerClient dockertools.DockerInterface
 	if s.ContainerRuntime == "docker" {
-		dockerClient = dockertools.CreateDockerClientOrDie(s.DockerEndpoint, s.RuntimeRequestTimeout.Duration)
+		dockerClient = dockertools.ConnectToDockerOrDie(s.DockerEndpoint, s.RuntimeRequestTimeout.Duration)
```

Comment:
There are ways to bypass this. E.g., we can add a variable `versionUpdated` in docker client, and check that before each call to make sure we've called `UpdateClientVersion` at least once. It'd add some more complexity.